### ISSUE TITLE
jpeg-recompress hangs on this image

### ISSUE THREAD

**Issue by biltongza** [danielgtaylor/jpeg-archive#99]
I use the Lightroom plugin from https://github.com/ftischhauser/JPEGOptimizer/ which uses jpeg-recompress. As part of my export process I have jpeg recompression enabled, and it is failing reliably on one of my images.

I caputed the command line it is using and tried running it myself, and it seems to just hang when I run it. In Lightroom, the recompression just fails immediately.

    jpeg-recompress.exe  --accurate --method smallfry --quality medium --strip --subsample disable --ppm - <path to image>

The image from Lightroom is large (50mb) so I couldn't attach it on GH, but here's a Google Drive link: https://drive.google.com/open?id=14LEd48_rgHMdC_5l_IR0VciLQ9xQ5zDe

If I CTRL+C the process, I get `jpeg-recompress: not a valid PPM format image!`, but otherwise it seems to just hang indefinitely.

**Comment by rsuzuki0** [comment]
I run a similar operation on Mac and observed the same problem since earlier version and with the current version. The problem only occurs with smallfry. The problem happens with maybe about one out of 20 images, exported from Lightroom.

**Comment by jlebar** [comment]
I think this is a dupe of #24.

I'm seeing if I can't figure it out to fix it, and if I can't, I will send a PR to add a warning to the docs about smallfry being broken.

**Comment by biltongza** [comment]
Yeah, it seems smallfry doesn't handle image dimensions that are not even. Even jpegmini crashed on it.
